controller/test/sensor_tests/sensors_test.cpp

```diff
@@ -291,3 +291,55 @@ TEST(SensorTests, Calibration) {
                   .cubic_m_per_sec(),
               COMPARISON_TOLERANCE_FLOW_CUBIC_M_PER_SEC);
 }
+
+TEST(SensorTests, FlowDrift) {
```

Comment:
This test is quite confusing to follow. Is it possible to have something simpler, like, simulate flow = k_signal * sin(t) + k_drift * sin(t / 10), and assert that the computed integral is close enough to the true integral?

Reply:
It would be tricky enough to come up with a signal and an error tolerance that today passes with the new code and fails with the old code.  I'm even more hesitant because if we ever change the PID, there's a decent chance the test will fail too...

Would it help if I wrote equivalents to `absl::c_max_element`?  For me that's a big part of what makes this ugly.